model-scoring-core/src/main/scala/org/trustedanalytics/scoring/ScoringServiceApplication.scala

```diff
@@ -74,36 +74,13 @@ class ScoringServiceApplication {
    * @return Model running inside the scoring engine instance
    */
   private def getModel: Model = {
-    var tempMarFile: File = null
-    try {
-      var marFilePath = config.getString("trustedanalytics.scoring-engine.archive-mar")
-      if (marFilePath.startsWith("hdfs:/")) {
-        if (!marFilePath.startsWith("hdfs://")) {
-          val relativePath = marFilePath.substring(marFilePath.indexOf("hdfs:") + 6)
-          marFilePath = "hdfs://" + relativePath
-        }
-
-        val hdfsFileSystem = try {
-          val token = new TapOauthToken(getJwtToken())
-          logger.info(s"Successfully retreived a token for user ${token.getUserName}")
-          Hdfs.newInstance().createFileSystem(token)
-        }
-        catch {
-          case t: Throwable =>
-            t.printStackTrace()
-            logger.info("Failed to create HDFS instance using hadoop-library. Default to FileSystem")
-            org.apache.hadoop.fs.FileSystem.get(new URI(marFilePath), new Configuration())
-        }
-        tempMarFile = File.createTempFile("model", ".mar")
-        hdfsFileSystem.copyToLocalFile(false, new Path(marFilePath), new Path(tempMarFile.getAbsolutePath))
-        marFilePath = tempMarFile.getAbsolutePath
-      }
+    val marFilePath = config.getString("trustedanalytics.scoring-engine.archive-mar")
+    if (marFilePath != "") {
       logger.info("calling ModelArchiveFormat to get the model")
-      sys.addShutdownHook(FileUtils.deleteQuietly(tempMarFile)) // Delete temporary directory on exit
       ModelArchiveFormat.read(new File(marFilePath), this.getClass.getClassLoader, None)
     }
-    finally {
-      FileUtils.deleteQuietly(tempMarFile)
+    else {
+      null
```

Comment:
consider using Option[Model] if returning null is acceptable.  It's more explicit.  This code requires null check for teh caller